test/jdk/javax/swing/JDesktopPane/bug4773378.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4773378
+ * @summary BasicDesktopPaneUI.desktopManager nulled after JDesktopPane.updateUI()
+ * @key headful
+ * @run main bug4773378
+ */
+
+import java.awt.AWTException;
+import java.awt.BorderLayout;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.beans.PropertyVetoException;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.DefaultDesktopManager;
+import javax.swing.JDesktopPane;
+import javax.swing.JFrame;
+import javax.swing.JInternalFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.event.InternalFrameAdapter;
+import javax.swing.event.InternalFrameEvent;
+
+public class bug4773378 {
+    JFrame frame;
+
+    JDesktopPane desktop;
+    DefaultDesktopManager desktopManager;
+    JInternalFrame jif;
+
+    Robot robot;
+    volatile boolean frameActivated = false;
+
+    public void setupGUI() {
+        frame = new JFrame("bug4773378");
+        frame.setLayout(new BorderLayout());
+        desktop = new JDesktopPane();
+        frame.add(desktop, BorderLayout.CENTER);
+
+        jif = new JInternalFrame("jif", true, false, true, true);
+        jif.setSize(150, 100);
+        jif.setVisible(true);
+        desktop.add(jif);
+
+        jif.addInternalFrameListener(new InternalFrameAdapter() {
+                public void internalFrameActivated(InternalFrameEvent e) {
+                    synchronized (bug4773378.this) {
+                        frameActivated = true;
+                        bug4773378.this.notifyAll();
+                    }
+                }
+            });
+
+        desktopManager = new MyDesktopManager();
+        desktop.setDesktopManager(desktopManager);
+        desktop.updateUI();
+
+        frame.setSize(200, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+        jif.requestFocus();
+    }
+
+    public void performTest() throws InterruptedException,
+            InvocationTargetException, AWTException {
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                jif.setSelected(true);
+            } catch (PropertyVetoException e) {
+                e.printStackTrace();
+            }
+        });
+
+        synchronized (this) {
+            while (!frameActivated) {
+                bug4773378.this.wait();
+            }
+        }
+
+        robot = new Robot();
+        robot.keyPress(KeyEvent.VK_CONTROL);
+        robot.keyPress(KeyEvent.VK_F6);
+        robot.keyRelease(KeyEvent.VK_F6);
+        robot.keyRelease(KeyEvent.VK_CONTROL);
+
+        Thread.sleep(2000);
```

Comment:
```suggestion
        robot.waitForIdle();
```
Is it really necessary to wait for 2 seconds before shutting down the test.

According to [JDK-4773378](https://bugs.openjdk.org/browse/JDK-4773378), `NullPointerException` was thrown when <kbd>Ctrl</kbd>+<kbd>F6</kbd> was pressed. The `waitForIdle` method doesn't return until the event queue is empty which implies the keyboard events are handled. It saves nearly 2 seconds.

Reply:
Ok.